Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java

```diff
@@ -0,0 +1,153 @@
+@Controller
+@Transactional(rollbackFor = { Exception.class })
+public class PublicBlogController {
+
+    @Autowired
+    private IProjectBlogDAO projectBlogDao;
+
+    @Autowired
+    private ProjectBlogDTOMapper projectBlogDTOMapper;
+
+    @Autowired
+    private IRetrieveProjectManager projectManager;
+
+    /**
+     * This method gives the the projectblog
+     * 
+     * @param projectUnixName
+     *            The project unix name
+     * @param model
+     *            Model
+     * @return view
+     * @throws QuadrigaStorageException
+     */
+    @RequestMapping(value = "sites/{projectUnixName}/projectblog", method = RequestMethod.GET)
+    public String projectblog(@PathVariable("projectUnixName") String projectUnixName, Model model)
+            throws QuadrigaStorageException {
+        IProject project = projectManager.getProjectDetailsByUnixName(projectUnixName);
+
+        if (project == null) {
+            return "forbidden";
+        }
+
+        // Creating Dummy Object
+        List dummyList = new ArrayList();
+        HashMap<String, String> map = new HashMap<String, String>();
+        map.put("title", "Article: Apple's iPhone Blunder");
+        map.put("text",
+                "<b> Can the United States </b> government compel Apple to help break into the phone of Syed Rizwan Farook, who, along with his wife Tafsheen Malil, gunned down fourteen innocent people last December at the Inland Regional Center in San Bernardino? That question has sparked fireworks in recent days. The dispute arises because Apple has equipped its new iPhones with encryption settings that erase the data contained on the phone whenever ten false password entries have been made. It was agreed on all sides that only Apple has the technology that might overcome the encryption device. [...]");
+        map.put("date", "February 22, 2016");
+        map.put("author", "Daniel T. Richards ");
+
+        dummyList.add(map);
+
+        model.addAttribute("blockentrylist", dummyList);
+        model.addAttribute("project", project);
+        model.addAttribute("project_id", project.getProjectId());
+
+        return "sites/projectblog";
+    }
+
+    /**
+     * This method provides page to add project blog entry to the authorized
+     * user. Only project owners are allowed to add entry to the blog.
+     * 
+     * @param projectUnixName
+     *            The project Unix name
+     * @param projectId
+     *            The id assigned to project by framework
+     * @param model
+     *            Model instance
+     * @return view to add new project blog entry
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     *             This exception is thrown when user other than project admin
+     *             and owner tries to add project blog entry.
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
```

Comment:
the add functionality should be in its own controller (roughly remember one controller per page).